packages/components/src/responsive-toolbar/index.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, cloneElement, Children, createRef } from '@wordpress/element';
+import { withInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import IconButton from '../icon-button';
+import Dropdown from '../dropdown';
+import Disabled from '../disabled';
+
+/**
+ * Module constants
+ */
+const OFFSET = 60;
+
+class ResponsiveToolbar extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			countHiddenChildren: 0,
+		};
+		this.container = createRef();
+		this.hiddenContainer = createRef();
+
+		this.throttledUpdateHiddenItems = this.throttledUpdateHiddenItems.bind( this );
+	}
+
+	componentDidMount() {
+		this.toggleWindowEvents( true );
+		this.updateHiddenItems();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.children !== this.props.children ) {
+			this.updateHiddenItems();
+		}
+	}
+
+	componentWillUnmount() {
+		this.toggleWindowEvents( false );
+	}
+
+	toggleWindowEvents( isListening ) {
+		const handler = isListening ? 'addEventListener' : 'removeEventListener';
+
+		window.cancelAnimationFrame( this.rafHandle );
+		window[ handler ]( 'resize', this.throttledUpdateHiddenItems );
+		window[ handler ]( 'scroll', this.throttledUpdateHiddenItems, true );
```

Comment:
Why `scroll` ?

Reply:
Probably useless, this was copied from the Popover but it's a different use-case there